src/protocol.rs

```diff
@@ -113,7 +116,12 @@ pub(crate) async fn read_bao_encoded<R: AsyncRead + Unpin>(
     reader: R,
     hash: Hash,
 ) -> Result<Vec<u8>> {
-    let mut decoder = AsyncSliceDecoder::new(reader, &hash.into(), 0, u64::MAX);
+    let mut decoder = bao_tree::r#async::AsyncResponseDecoder::new(
```

Comment:
this is an unfortunate module name

Reply:
yes, was going to ask you about this. Not entirely happy with it. Maybe `tokio_io`?

Reply:
Yeah `tokio_io` or `future_io` seems like a reasonable compromise. or sth like

```rust
mod io {
  mod sync {
  }
  mod tokio {
  }
}
```

Reply:
For now it is io and tokio_io. But this looks nice as well...